instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/OpenTelemetryAutoConfiguration.java

```diff
@@ -72,130 +47,37 @@ public static class OpenTelemetrySdkConfig {
-
-      return loggerProviderBuilder.build();
-    }
-
-    @Bean(destroyMethod = "") // SDK components are shutdown from the OpenTelemetry instance
-    @ConditionalOnMissingBean
-    public SdkMeterProvider sdkMeterProvider(
-        MetricExportProperties properties,
-        ObjectProvider<List<MetricExporter>> metricExportersProvider,
-        Resource otelResource) {
-
-      SdkMeterProviderBuilder meterProviderBuilder = SdkMeterProvider.builder();
-
-      metricExportersProvider.getIfAvailable(Collections::emptyList).stream()
-          .map(metricExporter -> createPeriodicMetricReader(properties, metricExporter))
-          .forEach(meterProviderBuilder::registerMetricReader);
-
-      return meterProviderBuilder.setResource(otelResource).build();
-    }
-
-    private static PeriodicMetricReader createPeriodicMetricReader(
-        MetricExportProperties properties, MetricExporter metricExporter) {
-      PeriodicMetricReaderBuilder metricReaderBuilder =
-          PeriodicMetricReader.builder(metricExporter);
-      if (properties.getInterval() != null) {
-        metricReaderBuilder.setInterval(properties.getInterval());
-      }
-      return metricReaderBuilder.build();
-    }
-
     @Bean
-    @ConditionalOnMissingBean
-    public Resource otelResource(
-        ConfigProperties config, ObjectProvider<List<ResourceProvider>> resourceProviders) {
-      Resource resource = Resource.getDefault();
-      for (ResourceProvider resourceProvider :
-          resourceProviders.getIfAvailable(Collections::emptyList)) {
-        resource = resource.merge(resourceProvider.createResource(config));
-      }
-      return resource;
+    public SpringComponentLoader springComponentLoader(ApplicationContext applicationContext) {
+      return new SpringComponentLoader(applicationContext);
     }
 
     @Bean
     // If you change the bean name, also change it in the OpenTelemetryJdbcDriverAutoConfiguration
     // class
     public OpenTelemetry openTelemetry(
-        ObjectProvider<ContextPropagators> propagatorsProvider,
-        SdkTracerProvider tracerProvider,
-        SdkMeterProvider meterProvider,
-        SdkLoggerProvider loggerProvider,
+        SpringComponentLoader componentLoader,
+        Optional<Resource> resource,
         ObjectProvider<List<OpenTelemetryInjector>> openTelemetryConsumerProvider) {
 
-      ContextPropagators propagators = propagatorsProvider.getIfAvailable(ContextPropagators::noop);
-
-      OpenTelemetrySdk openTelemetry =
-          OpenTelemetrySdk.builder()
-              .setTracerProvider(tracerProvider)
-              .setMeterProvider(meterProvider)
-              .setLoggerProvider(loggerProvider)
-              .setPropagators(propagators)
-              .build();
+      OpenTelemetry openTelemetry =
+          AutoConfiguredOpenTelemetrySdk.builder()
+              .setComponentLoader(componentLoader)
+              .addResourceCustomizer((given, config) -> resource.orElse(given))
```

Comment:
this is the only customizer that can be used instead of using the component loader